update how c++11 requirement is added

kejxu opened the pull request:
update how `c++11` compiler flag is added:
- only add it when it's supported (tested on Ubuntu 18.04)
- add similar logic for MSVC (as mentioned in the comment, this check would fail, but c++11 is always enabled). this pattern could be helpful if later this project moves to c++14 (which is an available switch in MSVC)

the other way to do this is to use `target_compile_features(mylib PUBLIC cxx_std_11)` from [cmake](https://cmake.org/cmake/help/v3.8/manual/cmake-compile-features.7.html#requiring-language-standards), but this would mean to add this line for every target. would this be preferred instead of adding a global `c++11` flag?

kejxu committed:
update compiler flag check (#9)

* update compiler flag check

* check for c++11 in MSVC and g++

kejxu committed:
add c++11 only when it's supported

tfoote commented:
If MSVC is using c++14 by default we shouldn't need to downgrade it for this. Can we just skip the declaration on Windows?

kejxu commented:
> If MSVC is using c++14 by default we shouldn't need to downgrade it for this. Can we just skip the declaration on Windows?

sure, this was mentioned during our porting too. would we want to keep `check_cxx_compiler_flag(-std=c++11 COMPILER_SUPPORTS_CXX11)` for G++ or change to just skip this check for MSVC?

tfoote commented:
We might as well improve the linux logic to check for it before applying it. And then just skip it on MSVC.

kejxu committed:
skip c++11 check for msvc

kejxu commented:
> We might as well improve the linux logic to check for it before applying it. And then just skip it on MSVC.

updated to skip `/std:c++11` check for msvc and updated comment
